sys-fs/ncdu/ncdu-2.2.2-r1.ebuild

```diff
@@ -0,0 +1,55 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit verify-sig edo
+
+DESCRIPTION="NCurses Disk Usage"
+HOMEPAGE="https://dev.yorhel.nl/ncdu"
+SRC_URI="
+	https://dev.yorhel.nl/download/${P}.tar.gz
+	verify-sig? ( https://dev.yorhel.nl/download/${P}.tar.gz.asc )
+"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+
+DEPEND="sys-libs/ncurses:=[unicode(+)]"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	|| ( dev-lang/zig:0.10 dev-lang/zig-bin:0.10 )
+	virtual/pkgconfig
+	dev-lang/perl
+	verify-sig? ( sec-keys/openpgp-keys-yorhel )
+"
+
+VERIFY_SIG_OPENPGP_KEY_PATH="${BROOT}"/usr/share/openpgp-keys/yoranheling.asc
+
+# see https://github.com/ziglang/zig/issues/3382
+# For now, Zig doesn't support CFLAGS/LDFLAGS/etc.
+QA_FLAGS_IGNORED="usr/bin/ncdu"
+
+src_unpack() {
+	if use verify-sig; then
+		verify-sig_verify_detached "${DISTDIR}"/${P}.tar.gz{,.asc}
+	fi
+	default
+}
+
+src_configure() {
+	local zig_version
+	zig_version="$(zig version)"
+	[[ ${zig_version} =~ 0.10.? ]] || die "You should choose Zig version 0.10.x via eselect-zig, found instead: $(zig version)"
```

Comment:
Couldn't the ebuild select automatically the correct zig version, instead of die'ing? See for example what the `egradle` function does in https://github.com/gentoo/gentoo/pull/28986/files#diff-62274241513f609709470a95a2d41d24f538e496f9415b2ef45553dfabfbab50R124.

Reply:
FTR, the actual logic that sets `EGRADLE` is in https://github.com/gentoo/gentoo/pull/28986/files#diff-62274241513f609709470a95a2d41d24f538e496f9415b2ef45553dfabfbab50R59.  Maybe this could be an inspiration for `EZIG`?

Reply:
Sounds very interesting. So create eclass (and send to gentoo-dev ofc), then ebuilds inherit this eclass and use `ezig` and f.i. `ZIG_SLOT`?

Reply:
Made it only for `ncdu` ebuild